packages/oauth/oauth-client/src/util.ts

```diff
@@ -116,3 +117,52 @@ export class CustomEventTarget<EventDetailMap extends Record<string, unknown>> {
     )
   }
 }
+
+export type SpaceSeparatedValue<Value extends string> =
+  | `${Value}`
+  | `${Value} ${string}`
+  | `${string} ${Value}`
+  | `${string} ${Value} ${string}`
+
+export const includesSpaceSeparatedValue = <Value extends string>(
+  input: string,
+  value: Value,
+): input is SpaceSeparatedValue<Value> => {
+  // Optimized version of:
+  // return input.split(' ').includes(value)
+
+  let idx = input.indexOf(value)
+  while (idx !== -1) {
+    const endIdx = idx + value.length
+    if (
+      // at beginning or preceded by space
+      (idx === 0 || input[idx - 1] === ' ') &&
+      // at end or followed by space
+      (endIdx === input.length || input[endIdx] === ' ')
+    ) {
+      return true
+    }
+    idx = input.indexOf(value, endIdx)
+  }
+
+  return false
+}
+
+export function combineSignals(
+  signals: readonly (AbortSignal | undefined)[],
+): undefined | AbortSignal {
+  const { length } = signals.filter(Boolean)
+  if (length === 0) return undefined
+  if (length === 1) return signals.find(Boolean)!
+
+  const controller = new AbortController()
+  const { signal } = controller
+
+  for (const signal of signals) {
+    if (signal) {
+      signal.addEventListener('abort', () => controller.abort(), { signal })
+    }
```

Comment:
There's shadowing going on here for `signal` which I think may cause some issues.  I assume the `{ signal }` options are intended to reference the outer `controller.signal`, to remove all listeners when `controller.abort()` occurs.